OpenRA.Game/Map/TileSet.cs

```
@@ -74,7 +74,10 @@ public TerrainTemplateInfo(TileSet tileSet, MiniYaml my)
 				{
 					int key;
 					if (!int.TryParse(node.Key, out key) || key < 0 || key >= tileInfo.Length)
-						throw new InvalidDataException("Invalid tile key '{0}' on template '{1}' of tileset '{2}'.".F(node.Key, Id, tileSet.Id));
+					{
+						Game.ModData.SpriteSequenceLoader.OnMissingSpriteError("Invalid tile key '{0}' on template '{1}' of tileset '{2}'.".F(node.Key, Id, tileSet.Id));
```

Why are these calling `ISpriteSequenceLoader.OnMissingSpriteError`? Tilesets have nothing to do with sequences, and these errors have nothing to do with sprites!

Note that #18514 will also be removing `OnMissingSpriteError` completely.

---

If you'd prefer not to come back to this PR I could take it over and add it as a new commit in #18514.

---

On closer inspection, I don't think these lines should change at all. Changing from an exception to silently dropping the template will introduce missing-file versus missing-sequence confusion that has confused modders with sequences for so long (and which i'm finally about to fix).

Can you please explain the motivation for ignoring the syntax errors here?

---

You suggested it in a review comment. The utility shouldn't crash if there are more errors ahead.

---

Ok. Can you verify that #18514 covers your use cases? If it does, we won't need to worry about this.